init.c

```diff
@@ -1905,26 +1909,35 @@ static void mutt_restore_default (struct option_t *p)
 #endif
 }
 
+static void ESC_char(char C, char* p, char *dst, size_t len)
+{
+  do
+  {
+     *p++ = '\\';
+     if (p - dst < len)
+       *p++ = C;
+  } while(0);
+}
+
 static size_t escape_string (char *dst, size_t len, const char* src)
```

Comment:
The whole point of the do { ... } while(0); construct is to turn the body of the macro into a block. By replacing the macro with a function this point fails to exist.